build-system/tasks/check-links.js

```diff
@@ -94,6 +99,22 @@ function checkLinks() {
 }
 
 /**
+ * Determines if a link points to a file added in the PR.
+ *
+ * @param {string} link Link being tested.
+ * @return {boolean} True if the link points to a file added in the PR.
+ */
+function isLinkToFileAddedInPR(link) {
+  var filesAdded = getStdout(
+      `git diff --name-only --diff-filter=A master...HEAD`).trim().split('\n');
+  var addedInPr = false;
+  filesAdded.forEach(function(file) {
+    if (file.length > 0 && link.includes(file)) addedInPr = true;
```

Comment:
`link` is any URL found in a `.md` file. For example, `[Foo](https://github.com/ampproject/amphtml/blob/master/contributing/foo.md)`. If `foo.md` were to be added by the same PR, the link checker will flag it as a dead link even though the link will be alive once the PR is merged.

The above function skips links that contain the name of a file being added by the PR.